ddtrace/internal/datadog/profiling/dd_wrapper/dd_wrapper.md

```diff
@@ -0,0 +1,74 @@
+# dd_wrapper
+
+This is experimental code that uses libdatadog for the collection and propagation of profiling data.  In theory this should just be a thin wrapper, but in practice some more state management.
+
+## Historical context
+
+Traditionally, dd-trace-py maintained three separate sampling mechanisms for the collection of profiling data.  These are called "collectors" in dd-trace-py parlance, but we'll refer to them as "samplers" in conformance with the broader litearture.
+
+### Stack sampler
+
+This is a simple stack sampler that collects stack traces at a dynamic interval.  Supposedly, it's a fixed-interval profiler, but the interval is adjusted dynamically depending on the observed overhead.  This sampler lives in a Python thread.  When it wakes up, it looks at the current threads/tasks and collects stack traces for each of them.  It also looks at the current exceptions and collects similar data for them.
```

Comment:
In a sense, yeah.  Note that although this is a description of the "v1" stack sampler, I do have support for propagating this context to echion in the echion PR.

What the noted method does is increase the amount of walltime between samples, which in turn associates more walltime (and almost definitely CPU time) to the subsequent sample.

It's a form of upscaling, but I don't prefer these instantaneous measures--they get really sloppy.  Having dynamically scaling sampling intervals all the time can wreak havoc on data fidelity, and I bet we introduce bias because the sampling rate becomes tightly coupled to some underlying behavior.

I had a vision to replace this mechanism with, at the very least, a PID controller.  That's TBD.